File: Frontend/types/analytics.ts
import type { SeverityStage } from "./photo";

export interface ProgressPoint {
  date: string;
  severity_stage: SeverityStage;
}

export interface Analytics {
  total_photos: number;
  latest_severity_stage: SeverityStage | null;
  progress: ProgressPoint[];
  avg_stress_level: number | null;
  avg_sleep_hours: number | null;
  active_treatments: number;
  treatment_names: string[];
}

File: Backend/app/presentation/schemas/analytics.py
from datetime import date

from pydantic import BaseModel


class ProgressPoint(BaseModel):
    date: date
    severity_stage: str | None
    confidence: float | None


class AnalyticsResponse(BaseModel):
    total_photos: int
    latest_severity_stage: str | None
    progress: list[ProgressPoint]
    total_habits_logged: int
    avg_stress_level: float | None
    avg_sleep_hours: float | None
    active_treatments: int
    treatment_names: list[str]
